### PR TITLE
fix: types

### LINKED ISSUE #65: TS Typing error: "This expression is not constructable." when esModuleInterop is false
# Bug Report

Hello, I am opening this issue as an follow-up to #16, which I believe still has issues with the TypeScript typing.
As I mentioned on my [comment](https://github.com/webpack-contrib/eslint-webpack-plugin/issues/16#issuecomment-763179893_), it is not possible to correctly import this package with its current type exports if I use the `esModuleInterop` set as false.

- Operating System:
- Node Version: v14.15.1
- NPM Version: 7.0.14
- webpack Version: 5.11.0
- eslint-webpack-plugin Version: 2.4.0
- eslint version: 7.7.0
- Typescript version: 4.0.1

### Expected Behavior

Importing "eslint-webpack-plugin" should correctly return the ESLintWebpackPlugin class and I should be able to instantiate it with the following code;
```
import ESLintWebpackPlugin = require("eslint-webpack-plugin");
import * as ESLintWebpackPlugin from "eslint-webpack-plugin"; // Alternative import.
import ESLintWebpackPlugin from "eslint-webpack-plugin" // when using esModuleInterop: true
...
new ESLintWebpackPlugin({
    // Options.
});
```

### Actual Behavior
Importing the default exports causes either a TS or a JS error when esModuleInterop is disabled, as I better explain on my comment [here](https://github.com/webpack-contrib/eslint-webpack-plugin/issues/16#issuecomment-763179893_).

### Code and Reproduction:
https://gist.github.com/fcaminada/7d2cc25c74c3b7bdfe43246190c5d57d

The solution would be to follow the same example as other plugins such as [CssMinimizerPlugin](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/css-minimizer-webpack-plugin/index.d.ts), for instance.

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
This PR contains a:

- [x] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Resolve #65

### CODE REVIEW

**Review by alexander-akait — COMMENTED** [review]
To be honestly we trying to avoid using `esModuleInterop` and `allowSyntheticDefaultImports`, this creates problems in some configuration

**Comment by ricardogobbosouza** [comment]
I will remove and correct possible problems

**Comment by ricardogobbosouza** [comment]
I will remove and correct possible problems

File: src/index.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { isAbsolute, join } from 'path';
 
+// @ts-ignore
 import arrify from 'arrify';
+// @ts-ignore
 import micromatch from 'micromatch';
```
Comment (alexander-akait) [inline src/index.js:6]: You can use `import * as micromatch from 'micromatch'`, it should help

Comment (ricardogobbosouza) [reply src/index.js:6]: fixed

**Comment by ArcanoxDragon** [comment]
This PR looks like it reverted most of #42, and as a result, I am experiencing the `This expression is not constructible` error _with_ `allowSyntheticDefaultImports` enabled now on `2.5.4`. I need `allowSyntheticDefaultImports` in my build-time `tsconfig.json` for other reasons.

**Pull request merged**